src/test/java/com/iota/iri/service/tipselection/impl/WalkerValidatorNoMaxDepthImplTest.java

```diff
@@ -0,0 +1,102 @@
+package com.iota.iri.service.tipselection.impl;
+
+import com.iota.iri.TransactionTestUtils;
+import com.iota.iri.conf.MainnetConfig;
+import com.iota.iri.conf.TipSelConfig;
+import com.iota.iri.controllers.TransactionViewModel;
+import com.iota.iri.model.Hash;
+import com.iota.iri.service.ledger.LedgerService;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+import com.iota.iri.service.snapshot.impl.SnapshotMockUtils;
+import com.iota.iri.service.tipselection.WalkValidator;
+import com.iota.iri.storage.Tangle;
+import com.iota.iri.storage.rocksDB.RocksDBPersistenceProvider;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(TransactionViewModel.class)
+public class WalkerValidatorNoMaxDepthImplTest {
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    private static final TemporaryFolder dbFolder = new TemporaryFolder();
+    private static final TemporaryFolder logFolder = new TemporaryFolder();
+    private static Tangle tangle;
+    private TipSelConfig config = new MainnetConfig();
+
+    @Mock
+    private static SnapshotProvider snapshotProvider;
+
+    @Mock
+    private static LedgerService ledgerService;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        tangle = new Tangle();
+        dbFolder.create();
+        logFolder.create();
+        tangle.addPersistenceProvider(new RocksDBPersistenceProvider(
+                dbFolder.getRoot().getAbsolutePath(), logFolder.getRoot().getAbsolutePath(), 1000,
+                Tangle.COLUMN_FAMILIES, Tangle.METADATA_COLUMN_FAMILY));
+        tangle.init();
+    }
```

Comment:
It is true that in many of our tests we have this code which uses a real DB.
However, I don't think this is a good practice...

There's lots of legacy code that you will find in IRI which is not good.

This holds especially truly now that you decided to add PowerMock...